### ISSUE TITLE
Wild Planter Box no longer works with recent update to ONI.

### ISSUE THREAD

**Issue by TroublesomeCorvid** [peteward44/ONIMods#10]
The game immediately crashes on startup.

**Comment by phschaad** [comment]
Same here, relevant part of the crash log:

```
[20:04:39.714] [1] [INFO] Loading mod content DLL [Wild Planter Box:2432751162] (provides DLL, Animation)
[20:04:39.716] [1] [INFO] Loading MOD dll: WildPlanterBox.dll
[20:04:39.904] [1] [INFO] [PLib] Mod WildPlanterBox initialized, version 1.0.0.0
[20:04:39.911] [1] [INFO] [PLib/WildPlanterBox] Registered mod options class PW.WildPlanterBoxOptions for WildPlanterBox
[20:04:39.911] [1] [WARNING] [PLibPatches] RegisterPatchClass could not find any handlers!
[20:04:39.915] [1] [ERROR] null Exception while loading mod WildPlanterBox at D:/UserData/Documents/Klei/OxygenNotIncluded/mods/Steam/2432751162. 
System.MissingMethodException: Method not found: void .Constraint..ctor(System.Func`2<KPrefabID, bool>,System.Func`2<Room, bool>,int,string,string,System.Collections.Generic.List`1<RoomConstraints/Constraint>)
  at PW.MyMod.OnLoad (HarmonyLib.Harmony harmony) [0x0006a] in <3b6e823b5afb4a829c7e7a0f9398e9fb>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00222] in <96bf83e479e841629616de31cacb5606>:0 
Exception while loading mod WildPlanterBox at D:/UserData/Documents/Klei/OxygenNotIncluded/mods/Steam/2432751162. 
System.MissingMethodException: Method not found: void .Constraint..ctor(System.Func`2<KPrefabID, bool>,System.Func`2<Room, bool>,int,string,string,System.Collections.Generic.List`1<RoomConstraints/Constraint>)
  at PW.MyMod.OnLoad (HarmonyLib.Harmony harmony) [0x0006a] in <3b6e823b5afb4a829c7e7a0f9398e9fb>:0 
  at KMod.DLLLoader.LoadDLLs (KMod.Mod ownerMod, System.String harmonyId, System.String path, System.Boolean isDev) [0x00222] in <96bf83e479e841629616de31cacb5606>:0 

....

[20:04:47.018] [1] [WARNING] [PLib/WildPlanterBox] System.IO.DirectoryNotFoundException Could not find a part of the path "D:\UserData\Documents\Klei\OxygenNotIncluded\mods\config\WildPlanterBox\config.json".   at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00164] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.IO.FileOptions options) [0x00000] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int,System.IO.FileOptions)
  at System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, System.Boolean detectEncodingFromByteOrderMarks, System.Int32 bufferSize) [0x00055] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at System.IO.StreamReader..ctor (System.String path, System.Boolean detectEncodingFromByteOrderMarks) [0x00007] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at System.IO.StreamReader..ctor (System.String path) [0x00000] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at (wrapper remoting-invoke-with-check) System.IO.StreamReader..ctor(string)
  at System.IO.File.OpenText (System.String path) [0x0000e] in <8ce0bd04a7a04b4b9395538239d3fdd8>:0 
  at PeterHan.PLib.Options.POptions.ReadSettings (System.String path, System.Type optionsType) [0x00002] in <3b6e823b5afb4a829c7e7a0f9398e9fb>:0 
[20:04:47.020] [1] [ERROR] First anim file needs to be non-null. WildFarmTileComplete (UnityEngine.GameObject)
First anim file needs to be non-null. WildFarmTileComplete (UnityEngine.GameObject)
[20:04:47.020] [1] [ERROR] null Exception in RegisterBuilding for type PW.WildFarmTileConfig from WildPlanterBox 
System.NullReferenceException: Object reference not set to an instance of an object
  at KAnimControllerBase.set_AnimFiles (KAnimFile[] value) [0x00046] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.Add2DComponents (BuildingDef def, UnityEngine.GameObject go, System.String initialAnimState, System.Boolean no_collider, System.Int32 layer) [0x00076] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.CreateBuildingComplete (UnityEngine.GameObject go, BuildingDef def) [0x000fe] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x000d8] in <96bf83e479e841629616de31cacb5606>:0 
  at (wrapper dynamic-method) GeneratedBuildings.GeneratedBuildings.LoadGeneratedBuildings_Patch3(System.Collections.Generic.List`1<System.Type>)
Exception in RegisterBuilding for type PW.WildFarmTileConfig from WildPlanterBox 
System.NullReferenceException: Object reference not set to an instance of an object
  at KAnimControllerBase.set_AnimFiles (KAnimFile[] value) [0x00046] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.Add2DComponents (BuildingDef def, UnityEngine.GameObject go, System.String initialAnimState, System.Boolean no_collider, System.Int32 layer) [0x00076] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.CreateBuildingComplete (UnityEngine.GameObject go, BuildingDef def) [0x000fe] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x000d8] in <96bf83e479e841629616de31cacb5606>:0 
  at (wrapper dynamic-method) GeneratedBuildings.GeneratedBuildings.LoadGeneratedBuildings_Patch3(System.Collections.Generic.List`1<System.Type>)
[20:04:47.021] [1] [WARNING] Missing Anim: [0x5D5E67FA]. You may have to run Collect Anim on the Assets prefab
[20:04:47.022] [1] [ERROR] First anim file needs to be non-null. WildPlanterBoxComplete (UnityEngine.GameObject)
First anim file needs to be non-null. WildPlanterBoxComplete (UnityEngine.GameObject)
[20:04:47.022] [1] [ERROR] null Exception in RegisterBuilding for type PW.WildPlanterBoxConfig from WildPlanterBox 
System.NullReferenceException: Object reference not set to an instance of an object
  at KAnimControllerBase.set_AnimFiles (KAnimFile[] value) [0x00046] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.Add2DComponents (BuildingDef def, UnityEngine.GameObject go, System.String initialAnimState, System.Boolean no_collider, System.Int32 layer) [0x00076] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.CreateBuildingComplete (UnityEngine.GameObject go, BuildingDef def) [0x000fe] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x000d8] in <96bf83e479e841629616de31cacb5606>:0 
  at (wrapper dynamic-method) GeneratedBuildings.GeneratedBuildings.LoadGeneratedBuildings_Patch3(System.Collections.Generic.List`1<System.Type>)
Exception in RegisterBuilding for type PW.WildPlanterBoxConfig from WildPlanterBox 
System.NullReferenceException: Object reference not set to an instance of an object
  at KAnimControllerBase.set_AnimFiles (KAnimFile[] value) [0x00046] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.Add2DComponents (BuildingDef def, UnityEngine.GameObject go, System.String initialAnimState, System.Boolean no_collider, System.Int32 layer) [0x00076] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingLoader.CreateBuildingComplete (UnityEngine.GameObject go, BuildingDef def) [0x000fe] in <96bf83e479e841629616de31cacb5606>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x000d8] in <96bf83e479e841629616de31cacb5606>:0 
  at (wrapper dynamic-method) GeneratedBuildings.GeneratedBuildings.LoadGeneratedBuildings_Patch3(System.Collections.Generic.List`1<System.Type>)
[20:04:47.023] [1] [WARNING] Mod: Building 'NaturalTile' was not added properly to PlanInfo, use ModUtil.AddBuildingToPlanScreen instead.
```

**Comment by peteward44** [comment]
Thanks all, this has been fixed in 4.0.3
